webapp/app/js/services/main.js

```diff
@@ -184,12 +184,24 @@ treeherder.factory('thNotify', [
             $log.debug("received message", message);
             var severity = severity || 'info';
             var sticky = sticky || false;
+            var maxNsNotifications = 5;
             thNotify.notifications.push({
                 message: message,
                 severity: severity,
                 sticky: sticky
             });
-            if(!sticky){
+
+            if (!sticky) {
+                if (thNotify.notifications.length > maxNsNotifications) {
+                    /* Replicate shift code to avoid pruning during rapid retriggers
+                     * otheriwse we could use $timeout(thNotify.shift) here */
+                    for(var i=0;i<thNotify.notifications.length; i++) {
```

Comment:
During testing with thNotify.shift(), the notifications will begin to not populate all 5 potential notification spots on screen (you will see 4, perhaps 3 notifications over a period of 15-30secs of flooding the retrigger 'queue'). If instead you execute the duplicated body code instead, it's fine and all 5 notification spots continue to cascade/populate on screen. I will try though what @camd suggests.